### ISSUE TITLE
MA0018 raised on `static abstract` or `static virtual` methods

### ISSUE THREAD

**Issue by Dryadxon** [meziantou/Meziantou.Analyzer#497]
**Context**
Rule identifier: _MA0018_
Target Framework: _.NET 7.0_
Language version: _11.0_

**Describe the bug**
The analyzer report MA0018 when `static abstract` or `static virtual` methods are defined inside generic interfaces.

**To Reproduce**
Trivial code example to reproduce the behavior:

````c#
public interface IFactory<TSelf> where TSelf : IFactory<TSelf>
{
    static abstract TSelf Create(); // Raise MA0018
}
````

**Describe the solution you'd like**
Rather than reporting MA0018, which should not apply to these cases, it would be better to raise a new error when `static abstract` and `static virtual` methods do not use the type arguments of these interfaces, and therefore suggest to:

- Make the interface non-generic, if no method uses type arguments.
- Divide the interface, into a non-generic one with all methods independent of type arguments, and a generic one that inherits from the first one, which defines all methods that use type arguments.

**Describe alternatives you've considered**
🤷 

**Additional context**
_none_

**Comment by meziantou** [comment]
I think the fix is to disable MA0018 for `static abstract`. I'll provide a fix soon.
